### ISSUE TITLE
1.1.0 problem with using seeds - no session

### ISSUE THREAD

**Issue by kzzzr** [ClickHouse/dbt-clickhouse#53]
Since you have released [1.1.0](https://github.com/ClickHouse/dbt-clickhouse/releases/tag/v1.1.0) it is no longer available to use command `dbt seed`.

```
05:52:19  Database Error in seed mytarget_campaigns (data/mytarget/mytarget_campaigns.csv)
05:52:19    :HTTPDriver url http://clickhouse:8123 returned response code 500)
05:52:19     b'Code: 113. DB::Exception: There is no session or session context has expired. (THERE_IS_NO_SESSION) (version 22.3.6.5 (official build))\n'
```

> By default all queries done via HTTP interface are stateless.
> To use temporary tables you need to have an active session. It's enough to add session_id parameter with some value to the URL of ClickHouse request. 

https://stackoverflow.com/questions/55758594/clickhouse-create-temporary-table

**Comment by kzzzr** [comment]
@guykoh what was the main reason to switch to `clickhouse_connect`?

**Comment by genzgd** [comment]
The new version should not use temporary tables in any circumstances, so a session should not be required.

**Comment by kzzzr** [comment]
Ok, then what am I doing wrong?

Dockerfile:
```Dockerfile
ARG DBT_VERSION=1.0.0
FROM fishtownanalytics/dbt:${DBT_VERSION}

ARG ADAPTER_POSTGRES_VERSION=1.1.0
RUN set -ex \
    && pip install dbt-postgres==${ADAPTER_POSTGRES_VERSION}

ARG ADAPTER_CLICKHOUSE_VERSION=1.1.0
RUN set -ex \
    && pip install dbt-clickhouse==${ADAPTER_CLICKHOUSE_VERSION}

WORKDIR /usr/app/
ENV DBT_PROFILES_DIR=.

ENTRYPOINT ["tail", "-f", "/dev/null"]
```

`docker-compose.yml`:
```
version: '2'
services:

  dbt:
    build: .
    volumes:
     - .:/usr/app/
    ports:
      - "8080:8080" 
    environment:
      DBT_SOURCE_DATABASE: default
      DBT_SOURCE_SCHEMA: default    

  clickhouse:
    image: clickhouse/clickhouse-server:22.3
    ports:
      - 9000:9000
      - 8123:8123
```

Then running commands:
```bash
docker-compose up --build -d
docker-compose exec dbt bash

root@3c5251946b60:/usr/app/integration_tests# dbt clean
13:40:14  Running with dbt=1.1.0
13:40:14  Checking target/*
13:40:15  Cleaned target/*
13:40:15  Checking dbt_packages/*
13:40:15  Cleaned dbt_packages/*
13:40:15  Finished cleaning all paths.
root@3c5251946b60:/usr/app/integration_tests# dbt deps
13:40:20  Running with dbt=1.1.0
13:40:20  Installing ../
13:40:20    Installed from <local @ ../>
root@3c5251946b60:/usr/app/integration_tests# dbt seed -s ads_adgroups
13:40:30  Running with dbt=1.1.0
13:40:30  Partial parse save file not found. Starting full parse.
13:40:36  Found 80 models, 526 tests, 0 snapshots, 0 analyses, 211 macros, 0 operations, 63 seed files, 80 sources, 0 exposures, 0 metrics
13:40:36  
13:40:36  Concurrency: 1 threads (target='ci_clickhouse')
13:40:36  
13:40:36  1 of 1 START seed file default.ads_adgroups .................................... [RUN]
HTTPDriver url http://clickhouse:8123 returned response code 500)
b'Code: 113. DB::Exception: There is no session or session context has expired. (THERE_IS_NO_SESSION) (version 22.3.6.5 (official build))\n'
13:40:36  1 of 1 ERROR loading seed file default.ads_adgroups ............................ [ERROR in 0.11s]
13:40:36  
13:40:36  Finished running 1 seed in 0.26s.
13:40:37  
13:40:37  Completed with 1 error and 0 warnings:
13:40:37  
13:40:37  Database Error in seed ads_adgroups (data/ads/ads_adgroups.csv)
13:40:37    :HTTPDriver url http://clickhouse:8123 returned response code 500)
13:40:37     b'Code: 113. DB::Exception: There is no session or session context has expired. (THERE_IS_NO_SESSION) (version 22.3.6.5 (official build))\n'
13:40:37  
13:40:37  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
root@3c5251946b60:/usr/app/integration_tests# 
```

**Comment by kzzzr** [comment]
Models are working.
Have you tested seeds by yourself? I can't find any unit test for seeds

**Comment by genzgd** [comment]
Are you using models that were generated under the previous `dbt-clickhouse` version?  (I'm not a dbt expert but I wonder if we have a backward compatibility issue).

**Comment by guykoh** [comment]
What do you mean by models are working? Have you tried running `dbt run` and it works? What do you see when you run `dbt debug`?

Almost all of the tests start with dbt seed command, so it is tested.

**Comment by guykoh** [comment]
@kzzzr, I ran your example with docker and verified that seed command works.

root@abf76da32317:/usr/app# dbt seed
15:47:42  Running with dbt=1.1.0
15:47:43  Unable to do partial parsing because profile has changed
15:47:51  Found 1 model, 0 tests, 1 snapshot, 0 analyses, 405 macros, 0 operations, 1 seed file, 6 sources, 0 exposures, 0 metrics
15:47:51  
15:47:51  Concurrency: 1 threads (target='dev')
15:47:51  
15:47:52  1 of 1 START seed file imdb_dbt.genre_codes .................................... [RUN]
15:47:52  1 of 1 OK loaded seed file imdb_dbt.genre_codes ................................ [INSERT 21 in 0.57s]
15:47:52  
15:47:52  Finished running 1 seed in 1.18s.
15:47:52  
15:47:52  Completed successfully
15:47:52  
15:47:52  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1

Maybe it relates to your profiles.yml file?

Here is my profiles.yml file:

clickhouse_imdb:
  target: dev
  outputs:
    dev:
      type: clickhouse
      schema: <target_db_name>
      host: clickhouse

Note: this assumes your username is 'default' and you run clickhouse without a password.

**Comment by kzzzr** [comment]
Oh yeah, I've figured out why this is happening.
I had pre-hook in recent versions that made it easier for me to insert csv data:

```
seeds:
  +pre-hook: "SET data_type_default_nullable = 1"
```

Is there any way to apply that setting for seeds right now?

**Comment by kzzzr** [comment]
Ok, somehow managed to COPY `users_settings.xml` inside container

**Comment by genzgd** [comment]
Can you explain how the lack of that setting is breaking your execution? I'm still trying to understand how you got the session error.

In any case, we should have an update shortly that fixes this issue by passing a session_id (as you originally suggested).  Thanks for digging into this and helping us understand and remedy the problem.

**Comment by guykoh** [comment]
Thanks @kzzzr and @genzgd, I released a new version of dbt-clickhouse (v1.1.0.1) with the pre-hook fixes.
@kzzzr you can update the connector now and test that seed command works as before - please lmk if there are any other issues with the seed command.
